test/python/test_deeplabv3p_ar_detect.py

```diff
@@ -0,0 +1,109 @@
+try:
+    from mpi4py import *
+    rank = MPI.COMM_WORLD.Get_rank()
+    n_ranks = MPI.COMM_WORLD.Get_size()
+except:
+    rank = 0
+    n_ranks = 1
+import os
+import sys
+import numpy as np
+from teca import *
+
+
+def get_padding_params(n_lat, n_lon):
+    target_shape = 128 * np.ceil(n_lat/128.0)
+    target_shape_diff = target_shape - n_lat
+    padding_amount_lat = (int(np.ceil(target_shape_diff / 2.0)), int(np.floor(target_shape_diff / 2.0)))
+
+    target_shape = 64 * np.ceil(n_lon/64.0)
+    target_shape_diff = target_shape - n_lon
+    padding_amount_lon = (int(np.ceil(target_shape_diff / 2.0)), int(np.floor(target_shape_diff / 2.0)))
+
+    return (padding_amount_lat, padding_amount_lon)
+
+
+def main():
+    set_stack_trace_on_error()
+    set_stack_trace_on_mpi_error()
+
+    if (len(sys.argv) != 7):
+        sys.stderr.write('\n\nUsage error:\n'\
+            'test_deeplabv3p_ar_detect [deeplab parameters] ' \
+            '[mesh data regex] [baseline mesh] [water vapor var] ' \
+            '[first step] [last step]\n\n')
+        sys.exit(-1)
+
+    # parse command line
+    deeplab_parameters = sys.argv[1]
+    mesh_data_regex = sys.argv[2]
+    baseline_mesh = sys.argv[3]
+    water_vapor_var = sys.argv[4]
+    first_step =  int(sys.argv[5])
+    last_step = int(sys.argv[6])
+
+    if (rank == 0):
+        sys.stderr.write('Testing on %d MPI processes\n'%(n_ranks))
+
+    cf_reader = teca_cf_reader.New()
+    cf_reader.set_files_regex(mesh_data_regex)
+    cf_reader.set_periodic_in_x(1)
+
+    # Getting padding parameters based on the lon & lat sizes
+    md = cf_reader.update_metadata()
+
+    try:
```

Comment:
do you need this try except? teca_metadata class already raising an exception, you are not handling it, but rather re-raising. I think that this try except could be removed reducing the amount of code without changing the runtime behavior.